File: rielarmoryangular/src/app/service/manufacturer.service.ts
import { Injectable } from '@angular/core';
import { Manufacturer } from '../model/manufacturer.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ManufacturerService {

private apiUrl = '/api/manufacturers';

  constructor(private http: HttpClient) { }

  getAllManufacturers(): Observable<Manufacturer[]> {
    return this.http.get<Manufacturer[]>(`${this.apiUrl}/all`);
  }
}

File: rielarmoryspringboot/src/main/java/com/gjbmloslos/rielarmoryspringboot/service/ManufacturerService.java
package com.gjbmloslos.rielarmoryspringboot.service;

import com.gjbmloslos.rielarmoryspringboot.model.Manufacturer;

import java.util.List;

public interface ManufacturerService {
    Manufacturer getManufacturerById(Long id);
    List<Manufacturer> getAllManufacturers();
}
